src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2242,6 +2248,49 @@ ostree_sysroot_write_deployments (OstreeSysroot     *self,
                                                         cancellable, error);
 }
 
+/* Finds the path to /boot on the filesystem that /boot is actually stored on.
+ * If you have a dedicated boot partition this will probably be "".  If it's
+ * stored on the root partition it will be "/boot".
+ *
+ * The path_out won't end with a "/".  That means that "/" is represented as ""
+ */
+static gboolean
+boot_find_path_on_disk(char** path_out, GError **error)
+{
+#ifdef HAVE_LIBMOUNT
+  g_autoptr(libmnt_table) tb = mnt_new_table();
+  g_assert (tb);
+  if (mnt_table_parse_mtab(tb, getenv("OSTREE_DEBUG_MOUNTINFO_FILE")) < 0)
+    return glnx_throw (error, "Failed to parse /proc/self/mountinfo");
+
+  const char *suffix = NULL;
+  struct libmnt_fs *fs = mnt_table_find_target(tb, "/boot", MNT_ITER_BACKWARD);
```

Comment:
TODO: Check for interactions with #1767. This might confuse `/boot` as bind-mount with `/boot` on own FS.

Also: This currently won't work for `ostree admin deploy --sysroot=xxx`.  Sometimes this is useful for embedded development where I'm deploying to an SD card and then plugging it into the embedded hardware for boot.

I've got some ideas on how to fix this.  I think I've implemented it in the past (in python), but I can't remember where.  I'll have a look.